src/pages/workspace/categories/WorkspaceCategoriesPage.tsx

```diff
@@ -220,31 +204,29 @@ function WorkspaceCategoriesPage({route}: WorkspaceCategoriesPageProps) {
         clearCategoryErrors(policyId, item.keyForList);
     };
 
-    const selectedCategoriesArray = Object.keys(selectedCategories).filter((key) => selectedCategories[key]);
-
     const handleDeleteCategories = () => {
-        setSelectedCategories({});
-        deleteWorkspaceCategories(policyId, selectedCategoriesArray);
+        setSelectedCategories([]);
+        deleteWorkspaceCategories(policyId, selectedCategories);
```

Comment:
Can we make sure that there are no `null` values in `selectedCategories`? We saw some cases where this was happening. I'm guessing it has to do with the logic on 160-170 above.

Reply:
I _think_ this is the PR that might have introduced the issue [here](https://github.com/Expensify/Expensify/issues/500746) though not sure cc @mountiny

Reply:
Yeah we should cover this here too! @daledah Could you please take a look and make sure the category list only includes non-empty strings and no other types? thanks!